### ISSUE TITLE
[MONITORING] Impact eco?

### ISSUE THREAD

**Issue by AudreyCLEVY** [etalab-ia/eg1#61]
@dtrckd 

Proposition: Use ecologits.ai 
in the EG1 evaluation to see the impact of our evaluation:

>>> https://ecologits.ai/
Environmental impacts are quantified based on four criteria and across two phases:

Criteria:

Energy (energy): Final energy consumption in kWh,
Global Warming Potential (gwp): Potential impact on global warming in kgCO2eq (commonly known as GHG/carbon emissions),
Abiotic Depletion Potential for Elements (adpe): Impact on the depletion of non-living resources such as minerals or metals in kgSbeq,
Primary Energy (pe): Total energy consumed from primary sources in MJ.
Phases:

Usage (usage): Represents the phase of energy consumption during model execution,
Embodied (embodied): Encompasses resource extraction, manufacturing, and transportation phases associated with the model's lifecycle.

**Comment by AudreyCLEVY** [comment]
ecologits does not allow Clientless initialization; (use litellm for simplification?)
But if we use Cortex, by default ecologits doesn't work without dev on our side

**Comment by AudreyCLEVY** [comment]
Use ComparIA method : from ecologits.tracers.utils import compute_llm_impacts, electricity_mixes
compatible with Albert API;

2 implementations:
- at response generation to get the "eco" info of the evaluated model
- at the observation part to measure the impact of llm evaluation

**Comment by AudreyCLEVY** [comment]
@dtrckd 

in Albert API, we now use aliases (excluding COrtex models): Albert-small and Albert-large;
This poses 2 “problems”:
- you lose the model name displayed in evals (albert-small doesn't matter which model is behind it);
- the notion of eco impact cannot be calculated

=> get the alias, and pass it as a parameter in the Exp (with alias_named that you've created)
aliases=['meta-llama/Llama-3.1-8B-Instruct'] e.g. for staging albert-small
==> on the front, if albert-small (or large) in the name, add the alias? (and thus adapt our current management)
==> for the eco impact part: take the alias when albert-small; otherwise take the name after the / (ex: mistralai/Mistral-Small-3.1-24B-Instruct-2503 ==> in the comparia conf file there's an alias; we'd have to modify them all to stick with the eg1 logic, which takes the whole name: 
(ex conf_compaia)
![Image](https://github.com/user-attachments/assets/62449208-a399-4795-b2b7-a21620c9429f)

What do you think? Let's talk about it Friday after the daily?

**Comment by dtrckd** [comment]
Hi, it seems that albert-api does still support the use of named model. I just test and it is still the case.
Do they plan to remove supporting named model ? 

If no, we just use named model for the evals.

**Comment by AudreyCLEVY** [comment]
Yes, now it will be done with the “alias” name, to simplify the maintainability of projects that run via albert API.

**Comment by dtrckd** [comment]
@leoguillaume @ArtixJP  can you confirm that we wont be able to use named model in the future ?

ps: this will make hard to track which model we evaluate on albert-api. Does the "aliases"  will remain though in the response of /models endpoint ?

**Comment by AudreyCLEVY** [comment]
for me, yes
now we have owned_by and aliasesby default
==> for example:
[Model(id='albert-small', created=1744728596, object='model', owned_by='Albert API', max_context_length=64000, type='text-generation', aliases=['meta-llama/Llama-3.1-8B-Instruct']),
